sdk/nodejs/iot/zMixins.ts

```diff
@@ -51,8 +51,11 @@ export class IoTHubEventSubscription extends appservice.EventSubscription<EventH
         // Place the mapping from the well known key name to the Event Hubs account connection string in
         // the 'app settings' object.
 
-        const appSettings = pulumi.all([args.appSettings, iotHub.eventHubEventsEndpoint]).apply(
-            ([appSettings, eventHubEventsEndpoint]) => ({ ...appSettings, [bindingConnectionKey]: eventHubEventsEndpoint }));
+        const appSettings = pulumi.all([args.appSettings, iotHub.eventHubEventsEndpoint, iotHub.sharedAccessPolicies]).apply(
+            ([appSettings, eventHubEventsEndpoint, sharedAccessPolicies]) => ({ 
+                ...appSettings, 
+                [bindingConnectionKey]: `Endpoint=${eventHubEventsEndpoint};SharedAccessKeyName=iothubowner;SharedAccessKey=${sharedAccessPolicies.filter(p => p.keyName === "iothubowner")[0].primaryKey}` 
```

Comment:
Is there a link to the official docs that we could put here, which would explain what's going on?

Reply:
Would this link help? https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-read-builtin

It describes how to get the "event hub compatible endpoint" connection string in the azure portal, and the code basically builds this string from the information available.

Reply:
Yes, I think it's helpful, thanks!

Reply:
done in 3de9fa2b988eb44505efbe34d1911977f28ef23b